alica_engine/include/engine/AlicaTimer.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include "engine/AlicaClock.h"
+#include "engine/IAlicaTimer.h"
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+#include <thread>
+
+namespace alicaTimer
```

Comment:
Seems not to match code style in other places

Also maybe we should call this DefaultAlicaTimer or AlicaSystemTimer

Reply:
This is old, I have just maintained, but yeah I agree should be snake case.

Reply:
yes but it is introduced in this PR and you are now the owner of the PR

Reply:
> Also maybe we should call this DefaultAlicaTimer or AlicaSystemTimer

This instead of SyncStopTimerTest?

Namespace should be `alica`, like all the rest.

Reply:
I mean the timer class itself, which is currently just AlicaTimer

Reply:
Besides the namespace being `AlicaTimer`, which I will change to `alica`, there's no class named like this inside the file. 
They are `SyncStopTimerImpl`, `SyncStopTimerTest`, `TimerFactory`.

`AlicaTime` is another class outside the scope of this PR, and it seems pretty old, so I assume it is correct. 

`IAlicaTimer` seems fine too, and after seeing this, `AlicaTimer` actually seems a better name than `SyncStopTimerTest`.

So, I propose to change from:
- `SyncStopTimerTest` to `AlicaTimer`,
- `SyncStopTimerImpl` to `AlicaTimerImpl`
- `TimerFactory` to `AlicaTimeFactory`
- namespace `AlicaTimer` to `alica`
- 
is it fine?

Reply:
Ah yeah certainly we should not name these after tests, but as per my earlier comment I would prefer renaming to AlicaSystemTimer or AlicaDefaultTimer rather than AlicaTimer

namespace change is fine